unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Scene/PerformanceTests/MessagingBusTest.cs

```diff
@@ -214,6 +72,128 @@ public void RemoveEntityShouldClearLossyMessages()
 
             Assert.AreEqual(0, bus.unreliableMessagesReplaced);
             Assert.AreEqual(3, bus.pendingMessagesCount);
+
+            return null;
+        }
+
+        [Test]
+        public void SceneMessageIsProcessedCorrectly()
+        {
+            //Given
+            QueuedSceneMessage.Type messageType = QueuedSceneMessage.Type.SCENE_MESSAGE;
+
+            QueuedSceneMessage_Scene queuedSceneMessageScene = new QueuedSceneMessage_Scene
+            {
+                payload = new Protocol.CreateEntity { entityId = entityId },
+                message = messageType.ToString(),
+                tag = "entity_1",
+                type = messageType
+            };
+
+            bus.Start();
+            bus.Enqueue(queuedSceneMessageScene);
+            //When
+            bus.ProcessQueue(0.1f, out _);
+
+            //Then
+            messageProcessHandler.Received(1).ProcessMessage(Arg.Any<QueuedSceneMessage_Scene>(), out Arg.Any<CustomYieldInstruction>());
```

Comment:
you can check `QueuedSceneMessage_Scene` data is correct with `Arg.Is<QueuedSceneMessage_Scene>(...)` by comparing it with `queuedSceneMessageScene`